### PR TITLE
fix(datetime): wheel picker font resizing works on android

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: resolves https://github.com/ionic-team/ionic-framework/issues/25964

This bug uncovered a few issues:

1. The `.picker-item` CSS in `picker-column-internal` assumed the container height would always be `34px`. When Android's font resize feature is used, the larger text is clipped because it overflows out of the `34px` tall container.
2. The `centerPickerItemInView` logic in `picker-column-internal` used a magic number for centering that also relied on each item being `34px` tall. When this changed, the centering logic broke causing the wrong item to be selected.
3. When fixing problem 2, it was discovered that the `.picker-item-empty` elements can have a different height that the non empty picker elements. The enabled items were `<button>` which defaulted to one font-family where the disabled items were `<div>` which defaulted to a different font-family. This resulted in different elements heights.

## What is the new behavior?

- Removed the `34px` assumption from the `.picker-item` declaration. The container now scales with the font size.
- Refactored the `centerPickerItemInView` logic by determining how much an element needs to be scrolled in order to reach the midpoint of the picker column container.
- Converted `.picker-item-empty` elements to `<button>`. Also added `disabled="true"` and `aria-hidden="true"` so they cannot receive focus or be announced by screen readers.

One design note: This `centerPickerItemInView` approach assumes that all buttons in the picker column are the same height. For now this is fine because long text truncates instead of wraps. If we expose the picker items as shadow parts in the future we may need to revisit this approach.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by liamdebeasi** [comment]
The screenshots are failing due to font family diffs. This is fixed by https://github.com/ionic-team/ionic-framework/pull/26020.

File: core/src/components/picker-column-internal/picker-column-internal.tsx [hunk 1]
```diff
@@ -228,10 +228,53 @@ export class PickerColumnInternal implements ComponentInterface {
...
+      const targetOffsetFromTop = (numberButton + 3) * targetBox.height;
+
+      /**
+       * Compute the scroll position necessary
+       * for the button to be centered within the view.
+       * Convert the value to an integer to avoid half
+       * pixel scrolls which can be inconsistent across
+       * browsers.
+       */
+      const top = Math.round(targetOffsetFromTop - midpoint);
```
Comment (amandaejohnston) [inline core/src/components/picker-column-internal/picker-column-internal.tsx:274]: https://user-images.githubusercontent.com/90629384/192820135-e2c14721-88fb-4aed-a157-eea30811f826.mp4 I'm seeing a weird issue where the time picker on the basic datetime opens with the wheels in the wrong positions, and then things jump around when I make the first click.

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

**Comment by liamdebeasi** [comment]
Going to move this back to a draft until I can figure out a better way to handle scrolling.

**Comment by liamdebeasi** [comment]
Closing for now until I can loop back to this.

**Pull request closed without merging**